src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -781,6 +790,20 @@ public Task<(long Index, object Result)> PutAsync(CommandBase cmd)
 
         public void SwitchToCandidateStateOnTimeout()
         {
+            using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                if (GetSnapshotRequest(context))
+                {
+                    // we aren't allowed to be elected for leadership if we requested a snapshot 
+                    if (Log.IsInfoEnabled)
+                    {
+                        Log.Info("we aren't allowed to be elected for leadership if we requested a snapshot");
+                    }
+                    return;
```

Comment:
Shouldn't we throw here?

Reply:
Like in `SetNewStateInTx` ?
Or is this because this is called from the timeout?

Reply:
yes, It is a timeout callback.